### PR TITLE
DG: Add delete feature

### PR DESCRIPTION
**Pull request by tllshan**
- Add implementation for delete in DG
- Add sequence diagrams for the parsing and execution of a delete command
image
image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,6 +158,25 @@ Classes used by multiple components are in the `wedlog.addressbook.commons` pack
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Delete feature
+
+The delete feature allows users to delete a guest or vendor in WedLog, through the respective classes `GuestDeleteCommand` and `VendorDeleteCommand`. The feature makes use of the current `Index` of the person in the displayed list to identify the person.
+
+Given below is an example usage scenario and how the delete mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. All guests and vendors are shown in their respective lists.
```
Comment (p-xp) [inline docs/DeveloperGuide.md:167]: Are we providing users with default guests and vendor values?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -184,6 +184,23 @@ A `TableNumber` object stores a table number as an integer. It is wrapped in an
   * Pros: Allows for greater flexibility in implementing features that are specific to either guests or vendors.
   * Cons: More code duplication.
 
+### Delete feature
+
+The delete feature allows users to delete a guest or vendor in WedLog, through the respective classes `GuestDeleteCommand` and `VendorDeleteCommand`. The feature makes use of the current `Index` of the person in the displayed list to identify the person.
+
+Given below is an example usage scenario and how the delete mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. All guests and vendors added during the last use of the app are shown in their respective lists.
```
Comment (p-xp) [inline docs/DeveloperGuide.md:193]: If they are using the application for the first time, they will not have any "last use of the app".
Perhaps scenario can be modified to be about user launching the application (not first time)

File: docs/diagrams/DeleteExecuteSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+activate XYZDeleteCommand
+
+XYZDeleteCommand -> Model : deleteXYZ(1)
+activate Model
+note right of XYZDeleteCommand: deleteXYZ = deleteGuest or deleteVendor
+
+Model -> UniqueGuestList : deleteXYZ(1)
+activate UniqueGuestList
+
+UniqueGuestList --> Model
```
Comment (p-xp) [inline docs/diagrams/DeleteExecuteSequenceDiagram.puml:30]: Should this be XYZ instead?

File: docs/diagrams/DeleteExecuteSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+deactivate Model
+
+create CommandResult
+XYZDeleteCommand -> CommandResult
+activate CommandResult
+
+CommandResult --> XYZDeleteCommand
+deactivate CommandResult
+
+XYZDeleteCommand --> LogicManager : result
```
Comment (p-xp) [inline docs/diagrams/DeleteExecuteSequenceDiagram.puml:43]: Should `XYZDeleteCommand` be deactivated for consistency?

**Review by p-xp — CHANGES_REQUESTED** [review]
LGTM, just a few nits!

File: docs/diagrams/DeleteExecuteSequenceDiagram.puml (hunk 4 above)
Comment (tllshan) [reply docs/diagrams/DeleteExecuteSequenceDiagram.puml:43]: oh cos the execution of the XYZCommand only happens in the following diagram, so I left it there for continuity

File: docs/diagrams/DeleteExecuteSequenceDiagram.puml (hunk 3 above)
Comment (tllshan) [reply docs/diagrams/DeleteExecuteSequenceDiagram.puml:30]: the method in Model is either deleteGuest or deleteVendor, so I put deleteXYZ to represent that

**Review by p-xp — APPROVED** [review]

**Pull request merged**